### PR TITLE
Better errors

### PR DESCRIPTION
**Pull request by bakkot**
This is on top of https://github.com/tc39/ecmarkup/pull/228 and is marked as a draft until that lands. https://github.com/tc39/ecmarkup/issues/226 remains open until a subsequent PR converts ecmarkdown failures into warnings.

This provides filename/line/column information for everything which is currently a warning, and unifies linting errors with other errors. The CLI wrapper for the library prints warnings out the same way it does currently (in red, as they are encountered), except with file/line/column. If you pass --verbose it also collects them at the end in sorted order using the provided eslint formatter.

This is a breaking change, at least if you consider ecmarkup to be a library and not just a CLI tool.

Screenshot (the 'FAIL: 1' is my prompt printing the exit status, not part of emarkup's output):

Screen Shot 2020-07-23 at 11 19 45 AM

<details>
<summary>screenshot without --strict</summary>
Screen Shot 2020-07-23 at 11 27 08 AM

</details>

### CODE REVIEW

File: src/Spec.ts [hunk 1]
```diff
@@ -408,7 +493,12 @@ export default class Spec {
     try {
       data = yaml.safeLoad(block.textContent!);
     } catch (e) {
```
Comment (michaelficarra) [inline src/Spec.ts:495]: Does this error contain position info we can preserve?

File: test/errors.js [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+      `,
+      {
+        ruleId: 'invalid-replacement',
+        nodeType: 'emu-alg',
+        message: 'expected algorithm to replace a step, not a clause',
+      }
+    );
+  });
+
+  it('cycle in replacement algorithms', async () => {
```
Comment (michaelficarra) [inline test/errors.js:70]: Can you add a bigger (non-self-referential) cycle test?

File: src/Spec.ts (hunk 1 above)
Comment (bakkot) [reply src/Spec.ts:495]: Technically yes, though it's not particularly good information. Updated.

File: test/errors.js (hunk 2 above)
Comment (bakkot) [reply test/errors.js:70]: Done.

**Pull request merged**